src/client/websockets.js

```diff
@@ -397,6 +397,10 @@ WebSocketManager.prototype.onMessageReceived = function (message, content, role)
 
     content = content || [];
     if (message !== '') {
+        // if the message is for requestId
+        stage.threads.processes.forEach(function (p) {
+            if (p.requestId === message) p.reply = msg;
```

Comment:
are you setting the message type to the request id? It would probably be better to use `msg.requestId` here